systemvm/patches/debian/config/opt/cloud/bin/cs_dhcp.py

```diff
@@ -34,16 +34,14 @@ def merge(dbag, data):
     return dbag
 
 
-def search(dbag, name):
+def search(dbag, name, gateway):
     """
     Dirty hack because CS does not deprovision hosts
     """
     hosts = []
     for o in dbag:
         if o == 'id':
             continue
-        print "%s %s" % (dbag[o]['host_name'], name)
-        if dbag[o]['host_name'] == name:
+        print "%s %s %s" % (dbag[o]['host_name'], name, gateway)
+        if dbag[o]['host_name'] == name and dbag[o]['default_gateway'] == gateway :
             hosts.append(o)
-    for o in hosts:
```

Comment:
@niteshsarda I tested with vm in 3 vpc tiers, however, it works fine, without the change on last 2 lines . We are running on a fork of cloudstack 4.7.1, so it might be fixed by other commits in our fork. need more investigation.